### PR TITLE
Piping Fixes For Shared Array Buffer Race Condition and Prompt Cancellation

### LINKED ISSUE #160: Stdin working funny
Stdin in the terminal is working in a funky/hacky way.
We should solve this.

This things should work with any browser:
* `cowsay hello | lolcat | lolcat` (it only works in FF now)
* `qjs | lolcat`

### PR DESCRIPTION
**Pull request by torch2424**
closes #160 

* Fixes race condition where Stdin would be piped to itself right before it died (`cowsay hi | lolcat | lolcat` on Desktop Chrome)
* Fixes Cancelling the prompt killing the entire process (`qjs | lolcat` on Firefox). 😄 
* Removes use of `npx` in build scripts, and adds CLI tools in the respective sub-packages

# Example

Screen Shot 2019-11-20 at 4 10 27 PM
Screen Shot 2019-11-20 at 4 06 07 PM

### CODE REVIEW

File: packages/wasm-terminal/src/process/process.ts [hunk 1]
```diff
@@ -114,16 +114,23 @@ export default class Process {
       }
 
       let error = "Unknown Error";
+      let isUserError = e.user !== undefined;
```
Comment (syrusakbary) [inline packages/wasm-terminal/src/process/process.ts]: Where `e.user` is defined? I'm a bit lost here

File: packages/wasm-terminal/src/process/process.ts [hunk 2]
```diff
@@ -114,16 +114,23 @@ export default class Process {
...
 
       if (e.code !== undefined) {
         error = `exited with code: ${e.code}`;
       } else if (e.signal !== undefined) {
         error = `killed with signal: ${e.signal}`;
-      } else if (e.user !== undefined) {
+      } else if (isUserError) {
         error = e.message;
```
Comment (syrusakbary) [inline packages/wasm-terminal/src/process/process.ts]: You can do directly here (removing new line 117 and changes after line 126)

```js
      } else if (e.user !== undefined) {
        end();
      }
      console.error(e);
      this.errorCallback(error, this.wasmFs.toJSON());
```

It makes the changes much easier to read (and less changes)

Comment (torch2424) [reply packages/wasm-terminal/src/process/process.ts]: Cool Sounds good :)

File: packages/wasm-terminal/src/process/process.ts (hunk 1 above)
Comment (torch2424) [reply packages/wasm-terminal/src/process/process.ts]: `e.user` is added as a boolean when the prompt is cancelled in `stdinRead( ... )`. It's just a flag to let us know the error is thrown by a user action, not an error in the wasm or something 😄

**Comment by torch2424** [comment]
@syrusakbary Made requested changes 😄 👍

**Pull request merged**